src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Extensions.Options;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.ServiceFabric.Client;
+using ServiceFabricCommon = Microsoft.ServiceFabric.Common;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class RestHealthCheckPublisher : HealthCheckPublisher
+	{
+		private readonly RestClientWrapper m_clientWrapper;
+
+		private IServiceFabricClient? m_client;
+
+		private readonly ILogger<RestHealthCheckPublisher> m_logger;
+
+		private string? m_nodeName;
+
+		internal override string HealthReportSourceId => nameof(RestHealthCheckPublisher);
+
+		internal const string NodeNameVariableName = "Fabric_NodeName";
+
+		public RestHealthCheckPublisher(ILogger<RestHealthCheckPublisher> logger,
+										IOptions<RestHealthCheckPublisherOptions> options,
+										ObjectPoolProvider objectPoolProvider) : base(objectPoolProvider)
```

Comment:
nit: we usually use a single tab for argumetns on a new line, please fix here and in other methods.